CMakeLists.txt

```diff
@@ -0,0 +1,98 @@
+cmake_minimum_required (VERSION 3.8.2)
+project(imgui VERSION 1.73.0 LANGUAGES CXX C)
+
+option (IMGUI_EXAMPLES "Build ImGui examples" ON)
```

Comment:
I think that it's better to have building any implementations or examples "OFF" by default, because this CMake build will have two types of users:
1. People who include ImGui as submodule, e.g. do `add_subdirectory(imgui-src)` and only want to link to imgui::imgui. It'll be pretty annoying to disable all the examples/backends if you don't need them
2. People who need the examples can set `IMGUI_EXAMPLES` to `ON` and only then all the possible implementations should be built

Reply:
It's better to have more lines in CI code than to force user to need to write more code for their use-case.

For example, consider the case where people want to build ImGui with SDL impl only. They'll need to opt-out of numerous implementations instead of opting-in for the implementations they want to build. E.g. I want to do:

```sh
cmake ../imgui-src -DIMGUI_IMPL_SDL=ON
```

instead of:

```sh
cmake ../imgui-src -DIMGUI_IMPL_GLFW=OFF -DIMGUI_IMPL_GLUT=OFF -DIMGUI_IMPL_OPENGL=OFF ...
```

And most users wouldn't want to build multiple/all implementations by default. 

As a compomise, we can have `IMGUI_BUILD_ALL_EXAMPLES` flag which will control if all possible implementations for the current OS should be built or not.  It still needs to be `OFF` by default, in my opinion.

Consider such examples of libraries which don't build their examples by default (and for many people all ImGui's implementations would be equal to "examples" if they don't want to use ImGui with something other than the framework of their choice)

[Clang](https://github.com/llvm/llvm-project/blob/main/clang/CMakeLists.txt#L517)
[SFML](https://github.com/SFML/SFML/blob/master/CMakeLists.txt#L66)
[GoogleTest](https://github.com/google/googletest/blob/master/googletest/CMakeLists.txt#L20)